shared/binlog.c

```diff
@@ -589,13 +595,20 @@ binlog * binlog_get_saved(binlog * node_binlog) {
 
 	/* free'd either here (binlog_destroy) or by the caller (node_binlog_read_saved) */
 	bl = malloc(sizeof(struct binlog));
+	if (bl == NULL) {
+		fclose(file);
+		return NULL;
+	}
+	memset(bl, 0, sizeof(struct binlog));
 
 	elements_read = fread(bl, sizeof(struct binlog), 1, file);
+	bl->path = strdup(node_binlog->file_metadata_path);
```

Comment:
Kindly provide info why this is necessary. 
These var seem to be allocated elsewhere in code. 
bl->path is allcoated in ln616 withe a different source.

Reply:
This is for the binlog_destroy call in line 611 so it can delete the binlog file which in this case is the meta file being read.

Reply:
Why not just use the bl->file_metadata_path? This would make it more clear what you are deleting. 
Alternatively, 
- you can move up line 615-616
OR 
- unlink the files prior to call to 611

Reply:
That was my initial thought before but the preceding line was opening/processing the meta file and the call to binlog_destroy unlinks what is was supposed to be in process. Succeeding lines also seems to reset the bl variable. It maybe more straight forward to do the unlink of the files outside the binlog_destroy call for this part as you suggest on the second alternate option.